### DISCUSSION TITLE
Report each workflow that the automatic GITHUB_TOKEN will not trigger instead of silently not triggering them

### DISCUSSION THREAD

**Discussion by Julian88Tex** [community/community#21091] [Actions]
### Situation

Currently, when using `GITHUB_TOKEN` in a GitHub Action that triggers other GitHub Actions, certain permissions may block those workflows from running. This isn't very clear and takes some research to understand why GitHub Actions are not running if you do not already understand `GITHUB_TOKEN` permissions. Specifically, GitHub Actions do not display in the run at all. For example, I'm only seeing 2 (run by a webhook and not GitHub action) here:

<img width="626" alt="Screen Shot 2022-07-21 at 2 55 52 PM" src="https://user-images.githubusercontent.com/459099/180514974-e406ac0b-cc84-4a20-9d9e-450f4c35f54e.png">

When I actually expect to see 5:

<img width="617" alt="Screen Shot 2022-07-21 at 2 55 48 PM" src="https://user-images.githubusercontent.com/459099/180515120-8b82a67a-e556-4a25-b407-dd4feea22369.png">

### Proposed Solution

It would be helpful if the non-running actions were displayed along with an error saying there is a permissions issue. Alternatively, there could be an error or warning on the GitHub Action which is triggering the other actions saying that it doesn't have permission to trigger.

**Comment by jsoref** [comment]
Do you have a particular example?

My example is https://github.com/actions/labeler/issues/136#issuecomment-1169411989, which, simply put, is a bug:

https://github.com/check-spelling/tokio/runs/7491144748?check_suite_focus=true

```
Current runner version: '2.294.0'
Operating System
Virtual Environment
Virtual Environment Provisioner
GITHUB_TOKEN Permissions
  Contents: read
  Metadata: read
Secret source: Actions
Prepare workflow directory
Prepare all required actions
Getting action download info
Download action repository 'actions/labeler@v3' (SHA:3d61[2](https://github.com/check-spelling/tokio/runs/7491144748?check_suite_focus=true#step:1:2)d72e[6](https://github.com/check-spelling/tokio/runs/7491144748?check_suite_focus=true#step:1:7)784a1a65365cc6d33b5a001c12bf10)
```

```
Run actions/labeler@v3
  with:
    repo-token: ***
    sync-labels: true
    configuration-path: .github/labeler.yml
Error: HttpError: Resource not accessible by integration
Error: Resource not accessible by integration
```

If an action doesn't handle permissions errors, those are bugs, and they should be reported.

As an action author, I can say that our code can be just as buggy as everyone else's code. (I've certainly had permissions errors in my code, although I'm slowly working on stamping them out.)

That said, I think I may have even made a request somewhere that they improve the error report from `Resource not accessible by integration` to include which permission(s) are needed.

Sadly, doing that would break existing code, such as:
https://github.com/tanmen/jest-reporter/blob/b27b26dc15223cdcff8023aeba8dda729a9ca34e/src/report.ts#L43

**Reply by Julian88Tex** [reply]
@jsoref Thanks for that additional context. I could provide more details on a GitHub Action, but I'd say that this request is for the GitHub Actions team to have the build fail regardless of whether or not there is the catching of this error by the action itself.

**Comment by jsoref** [comment]
Fwiw, are you sure that this is the error you're hitting? (Really, a URL would be really helpful.)

There's a recent-ish feature where PRs from non-members won't run GitHub actions w/o approvals. This isn't precisely a `GITHUB_TOKEN` permission problem, as once authorized, the workflows would run.

**Comment by Julian88Tex** [comment]
@jsoref hey sure. So that repo where I originally saw the issue is private so I created a test one. For this repo I'm using an [autoupdate](https://github.com/chinthakagodawita/autoupdate) action to trigger builds merged from one branch `release-1` automatically into `release-2` which has a [PR](https://github.com/Julian88Tex/autoupdate-token-test/pull/1).

You'll notice on this PR that sometimes github action builds run automatically and sometimes they do no. I've been switching the token that autoupdate uses back and forth from a PAT to `GITHUB_TOKEN`. When it's a `PAT` the github action builds like "hello world" run just fine, but when I use the `GITHUB_TOKEN` those builds do not run with no indication.

I've added you as a collaborator if you want to give it a try :).

**Reply by jsoref** [reply]
Oh, this is just [the `GITHUB_TOKEN` doesn't trigger workflows](https://docs.github.com/en/actions/using-workflows/triggering-a-workflow#triggering-a-workflow-from-a-workflow). It's by design.

**Reply by Julian88Tex** [reply]
@jsoref thanks for that official documentation and I figured it was by design. Feature request here is make this obvious whenever it happens. Sounds like it should be a warning whenever a GitHub Action attempts to trigger a workflow.

**Reply by jsoref** [reply]
Hmm, that's a pretty reasonable request.

As an implementation detail, it literally _doesn't try_ to trigger a workflow, but yeah, it seems reasonable for it to post a thing saying "Hey, github-actions isn't triggering a workflow, if you wanted this to, see ..." -- obviously there's a code branch for it to think about, and putting some reporting in at that point shouldn't be hard.

(Admittedly, I've never seen the code...)

Fwiw, I'd change the summary of this discussion to:

> Report each workflow that the automatic `GITHUB_TOKEN` will not trigger instead of silently not triggering them

**Reply by Julian88Tex** [reply]
@jsoref Appreciate that recommendation and done!
